setup.py

```diff
@@ -0,0 +1,122 @@
+import io
+import re
+import os
+import subprocess
+from pkg_resources import parse_requirements
+from setuptools import find_packages, setup
+from setuptools import setup, find_packages, Command
+from setuptools.command.bdist_egg import bdist_egg as _bdist_egg
+from setuptools.command.develop import develop as _develop
+from distutils.command.build import build as _build
+
+SETUP_DIR = os.path.dirname(os.path.abspath(__file__))
+
+class fetch_guacamol_datasets(Command):
+    """
+    Run installation to fetch guacamol datasets.
+    """
+
+    def initialize_options(self):
+        """Set initialize options."""
+        pass
+
+    def finalize_options(self):
+        """Set finalize options."""
+        pass
+
+    def run(self):
+        """Run installation to fetch guacamol datasets."""
+        try:
+            subprocess.check_call(
+                [
+                    os.path.join(SETUP_DIR, 'fetch_guacamol_dataset.sh')
+                ]
+            )
+        except subprocess.CalledProcessError as error:
+            raise EnvironmentError(
+                f"Failed to fetch of guacamol datasets dependencies via {error.cmd}."
+            )
+        
+
+class build(_build):
+    """Build command."""
+
+    sub_commands = [("fetch_guacamol_datasets", None)] + _build.sub_commands
+
+
+class bdist_egg(_bdist_egg):
+    """Build bdist_egg."""
+
+    def run(self):
+        """Run build bdist_egg."""
+        self.run_command("fetch_guacamol_datasets")
+        _bdist_egg.run(self)
+
+
+class develop(_develop):
+    """Build develop."""
+
+    def run(self):
+        """Run build develop."""
+        setup_cosifer = self.distribution.get_command_obj("fetch_guacamol_datasets")
+        setup_cosifer.develop = True
+        self.run_command("fetch_guacamol_datasets")
+        _develop.run(self)
+
+
+# ease installation during development
+vcs = re.compile(r"(git|svn|hg|bzr)\+")
+try:
+    with open("dockers/requirements.txt") as fp:
+        VCS_REQUIREMENTS = [
+            str(requirement)
+            for requirement in parse_requirements(fp)
+            if vcs.search(str(requirement))
+        ]
+except FileNotFoundError:
+    # requires verbose flags
+    print("requirements.txt not found.")
+    VCS_REQUIREMENTS = []
+
+match = re.search(
+    r'__version__\s*=\s*[\'"]([^\'"]*)[\'"]',
+    io.open("guacamol_baselines/__init__.py", encoding="utf_8_sig").read(),
+)
+if match is None:
+    raise SystemExit("Version number not found.")
+__version__ = match.group(1)
+
+setup(
```

Comment:
also make sure we add the data, once moved in the package folder, as package data to be tracked by `pkg_resources` (see example: https://github.com/IBM/mimkl/blob/53a5a9db5aa09c6e8808ba5b845601c5768d23e2/setup.py#L137)